apps/smart-contracts/core/test/DepositHook.test.ts

```diff
@@ -342,4 +369,96 @@ describe('=> DepositHook', () => {
       await expect(tx).to.emit(depositHook, 'AllowlistChange').withArgs(mockAllowlist.address)
     })
   })
+
+  describe('# setCollectionScores', () => {
+    it('reverts if not role holder', async () => {
+      expect(
+        await depositHook.hasRole(await depositHook.SET_COLLECTION_SCORES_ROLE(), user.address)
+      ).to.eq(false)
+
+      await expect(depositHook.connect(user).setCollectionScores([], [])).revertedWith(
+        `AccessControl: account ${user.address.toLowerCase()} is missing role ${await depositHook.SET_COLLECTION_SCORES_ROLE()}`
+      )
+    })
+
+    it('reverts if collections.length != scores.length', async () => {})
+
+    it('reverts if collection is added with 0 score', async () => {})
+
+    it('sets score for one collection', async () => {})
+
+    it('sets score for multiple collections', async () => {})
+
+    it('sets score for same collection twice', async () => {})
+
+    it('is idempotent', async () => {})
+
+    it('emits CollectionScoresChange', async () => {})
+  })
+
+  describe('# removeCollections', () => {
```

Comment:
```suggestion
  describe('# removeCollections', () => {
```
This test block as well.